src/main/java/org/testng/reporters/FailedReporter.java

```diff
@@ -54,84 +55,50 @@ protected void generateFailureSuite(XmlSuite xmlSuite, ISuite suite, String outp
     failedSuite.setName("Failed suite [" + xmlSuite.getName() + "]");
     m_xmlSuite= failedSuite;
 
-    Map<String, XmlTest> xmlTests= Maps.newHashMap();
-    for(XmlTest xmlT: xmlSuite.getTests()) {
-      xmlTests.put(xmlT.getName(), xmlT);
-    }
-
     Map<String, ISuiteResult> results = suite.getResults();
 
-    synchronized(results) {
```

Comment:
@juherr - Here `results` is a local reference to a Map. How would synchronization on a local variable help ? Can you please help me understand that ? 

Also I believe we don't need synchronization here (if we do, then we may need to add a sync block on either this or add an explicit mutex) because :
* The method `org.testng.reporters.FailedReporter#generateFailureSuite` is called as part of `generateReport`.
* TestNG does not run any listeners in parallel, so there is no way in which the map returned by `org.testng.SuiteRunner#getResults` is going to be altered by someone else.

Please advise if I am missing something.